Gosu/Input.hpp

```diff
@@ -92,29 +94,34 @@ namespace Gosu
         // Undocumented for the moment. Also applies to current_touches().
         void set_mouse_factors(double factor_x, double factor_y,
             double offset_x = 0, double offset_y = 0);
-        
+
         //! Currently known touches.
         const Touches& current_touches() const;
-        
+
         //! Accelerometer positions in all three dimensions (smoothened).
         //! Note: Not really deprecated, but temporarily defunct.
         GOSU_DEPRECATED double accelerometer_x() const;
         GOSU_DEPRECATED double accelerometer_y() const;
         GOSU_DEPRECATED double accelerometer_z() const;
-        
+
         //! Collects new information about which buttons are pressed, where the
         //! mouse is and calls on_button_up/on_button_down, if assigned.
         void update();
-        
+
         //! Assignable events that are called by update. You can bind these to your own functions.
         //! If you use the Window class, it will assign these to its own methods.
         std::function<void (Button)> on_button_down, on_button_up;
-        
+
+        std::function<void (Button, double)> on_axis_motion;
+
+        std::function<void (int, std::string)> on_gamepad_connected;
```

Comment:
I added `gamepad_name(id)`. It works from within the `gamepad_disconnected` callback since the slot isn't cleared until after the callback is called.

I also added the `button_name(id)` function, I assume its purpose is to return the name for the id? `button_name(Gosu::KbSpace) => "Space"` (TODO: return in lowercase)